server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -312,68 +314,60 @@ private void addComponentRelatedFilters(IssueQuery query, Map<String, FilterBuil
   }
 
   @CheckForNull
-  private FilterBuilder createViewFilter(Collection<String> viewUuids) {
+  private QueryBuilder createViewFilter(Collection<String> viewUuids) {
     if (viewUuids.isEmpty()) {
       return null;
     }
 
-    OrFilterBuilder viewsFilter = FilterBuilders.orFilter();
+    BoolQueryBuilder viewsFilter = boolQuery();
     for (String viewUuid : viewUuids) {
-      viewsFilter.add(FilterBuilders.termsLookupFilter(IssueIndexDefinition.FIELD_ISSUE_PROJECT_UUID)
+      viewsFilter.should(QueryBuilders.termsLookupQuery(IssueIndexDefinition.FIELD_ISSUE_PROJECT_UUID)
         .lookupIndex(ViewIndexDefinition.INDEX)
         .lookupType(ViewIndexDefinition.TYPE_VIEW)
         .lookupId(viewUuid)
-        .lookupPath(ViewIndexDefinition.FIELD_PROJECTS))
-        .cacheKey(viewsLookupCacheKey(viewUuid));
+        .lookupPath(ViewIndexDefinition.FIELD_PROJECTS));
     }
     return viewsFilter;
   }
 
-  public static String viewsLookupCacheKey(String viewUuid) {
-    return String.format("%s%s%s", IssueIndexDefinition.TYPE_ISSUE, viewUuid, ViewIndexDefinition.TYPE_VIEW);
-  }
-
-  private static FilterBuilder createAuthorizationFilter(boolean checkAuthorization, @Nullable String userLogin, Set<String> userGroups) {
+  private static QueryBuilder createAuthorizationFilter(boolean checkAuthorization, @Nullable String userLogin, Set<String> userGroups) {
     if (checkAuthorization) {
-      OrFilterBuilder groupsAndUser = FilterBuilders.orFilter();
+      BoolQueryBuilder groupsAndUser = boolQuery();
       if (userLogin != null) {
-        groupsAndUser.add(FilterBuilders.termFilter(IssueIndexDefinition.FIELD_AUTHORIZATION_USERS, userLogin));
+        groupsAndUser.should(termQuery(IssueIndexDefinition.FIELD_AUTHORIZATION_USERS, userLogin));
       }
       for (String group : userGroups) {
-        groupsAndUser.add(FilterBuilders.termFilter(IssueIndexDefinition.FIELD_AUTHORIZATION_GROUPS, group));
+        groupsAndUser.should(termQuery(IssueIndexDefinition.FIELD_AUTHORIZATION_GROUPS, group));
       }
-      return FilterBuilders.hasParentFilter(IssueIndexDefinition.TYPE_AUTHORIZATION,
+      return QueryBuilders.hasParentQuery(IssueIndexDefinition.TYPE_AUTHORIZATION,
         QueryBuilders.filteredQuery(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "filteredQuery"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)